### PR TITLE
Escape CSI and K_SPECIAL in ins_char_typebuf()

### PR DESCRIPTION
**Pull request by zeertzjq**

### CODE REVIEW

**Comment by zeertzjq** [comment]
Hmm, I'm a bit confused by the #ifdefs in `fix_input_buffer()`.

**Comment by zeertzjq** [comment]
And CSI seems to also need escaping in GUI...

File: src/getchar.c [hunk 1]
```diff
@@ -1142,8 +1142,9 @@ ins_char_typebuf(int c, int modifier)
     }
     else
     {
-	len += (*mb_char2bytes)(c, buf + len);
-	buf[len] = NUL;
+	char_u *p = buf + len;
+	int char_len = (*mb_char2bytes)(c, p);
+	len += fix_input_buffer(p, char_len);
```
Comment (yegappan) [inline src/getchar.c:1152]: Can you add some comment on why this is needed? Why CSI and K_SPECIAL need to be escaped?

Comment (zeertzjq) [reply src/getchar.c:1152]: Some characters like "固", "四", "倒" are not inserted when typed in Select mode.

**Comment by zeertzjq** [comment]
I'm trying to remove some conditions and see if test passes on Windows.

File: src/getchar.c (hunk 1 above)
Comment (zeertzjq) [reply src/getchar.c:1152]: I'll first wait for CI, and add comments later.

**Comment by zeertzjq** [comment]
Oh, the condition is still needed...

**Comment by zeertzjq** [comment]
Surprisingly all checks passed if I remove the `#ifndef MSWIN`. Is this because there are no tests that are depend on this, or that the `#ifndef MSWIN` added in patch 8.1.1230 is really not needed?

**Pull request closed without merging**